### ISSUE TITLE
Cannot connect to server

### ISSUE THREAD

**Issue by Azesuper** [QuiltMC/quilt-loader#354]
The server is running normally but no one can connect to it there’s no mods and I already port forwarded other servers running on my computer work like fabric or vanilla but it’s just quilt that doesn’t work

**Comment by Azesuper** [comment]
I’m using 1.19.2

**Comment by Azesuper** [comment]
I followed the instructions on the quilt page to download the server file

**Comment by Azesuper** [comment]
It loaded overworld already and seems to work fine but nothing can even sense it online

**Comment by Player505** [comment]
I'm getting same issue on 1.20.1

```
23:12:49] [Server Connector #1/ERROR]: Couldn't connect to server
io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: minecraft-technical.tail109a4.ts.net/100.64.204.54:25565
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:359) ~[netty-transport-native-unix-common-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:489) ~[netty-transport-classes-epoll-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[netty-transport-classes-epoll-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

**Comment by LuminaSapphira** [comment]
Same issue here - tried with loader versions 0.20.1, 0.20.0, and 0.19.5. 0.20+ fails while 0.19.5 works fine. Stock installation, no mods on either server or client.

OS in case it matters: `Arch Linux - 6.4.11-zen2-1-zen`

**Comment by Azesuper** [comment]
How has this issue not been reported before

**Comment by AlexIIL** [comment]
From testing locally `0.20.0-beta.5` is the most recent version that worked (0.20.0-beta.6 has this bug), which means the issue was introduced in [this commit](https://github.com/QuiltMC/quilt-loader/commit/084f11b8c3606327899e14094faaef846f2ebfbb) (range: https://github.com/QuiltMC/quilt-loader/compare/0.20.0-beta.5...0.20.0-beta.6)

**Comment by Azesuper** [comment]
Ok so use 0.20.0-beta 5 and it will work right?

**Comment by Azesuper** [comment]
Thanks

**Comment by AlexIIL** [comment]
Enabling [debug logging](https://wiki.vg/Debugging#All_debug_information) logged this stacktrace when I tried to connect:
```java
[15:25:57] [Netty Epoll Server IO #1/DEBUG]: Enabled auto read
[15:25:57] [Netty Epoll Server IO #1/DEBUG]: Failed to sent packet
java.lang.ExceptionInInitializerError: null
	at io.netty.buffer.PoolThreadCache.createSubPageCaches(PoolThreadCache.java:117) ~[netty-buffer-4.1.82.Final.jar:?]
	at io.netty.buffer.PoolThreadCache.<init>(PoolThreadCache.java:74) ~[netty-buffer-4.1.82.Final.jar:?]
	at io.netty.buffer.PooledByteBufAllocator$PoolThreadLocalCache.initialValue(PooledByteBufAllocator.java:526) ~[netty-buffer-4.1.82.Final.jar:?]
	at io.netty.buffer.PooledByteBufAllocator$PoolThreadLocalCache.initialValue(PooledByteBufAllocator.java:503) ~[netty-buffer-4.1.82.Final.jar:?]
	at io.netty.util.concurrent.FastThreadLocal.initialize(FastThreadLocal.java:177) ~[netty-common-4.1.82.Final.jar:?]
	at io.netty.util.concurrent.FastThreadLocal.get(FastThreadLocal.java:142) ~[netty-common-4.1.82.Final.jar:?]
	at io.netty.buffer.PooledByteBufAllocator.newDirectBuffer(PooledByteBufAllocator.java:391) ~[netty-buffer-4.1.82.Final.jar:?]
	at io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:188) ~[netty-buffer-4.1.82.Final.jar:?]
	at io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:179) ~[netty-buffer-4.1.82.Final.jar:?]
	at io.netty.channel.unix.PreferredDirectByteBufAllocator.ioBuffer(PreferredDirectByteBufAllocator.java:53) ~[netty-transport-native-unix-common-4.1.82.Final.jar:?]
	at io.netty.channel.DefaultMaxMessagesRecvByteBufAllocator$MaxMessageHandle.allocate(DefaultMaxMessagesRecvByteBufAllocator.java:120) ~[netty-transport-4.1.82.Final.jar:?]
	at io.netty.channel.epoll.EpollRecvByteAllocatorHandle.allocate(EpollRecvByteAllocatorHandle.java:75) ~[netty-transport-classes-epoll-4.1.82.Final.jar:?]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:785) ~[netty-transport-classes-epoll-4.1.82.Final.jar:?]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499) ~[netty-transport-classes-epoll-4.1.82.Final.jar:?]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[netty-transport-classes-epoll-4.1.82.Final.jar:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.82.Final.jar:?]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.82.Final.jar:?]
	at java.lang.Thread.run(Thread.java:1589) ~[?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 6 out of bounds for length 6
	at org.objectweb.asm.ClassReader.readShort(ClassReader.java:3623) ~[asm-9.5.jar:9.5]
	at org.objectweb.asm.ClassReader.<init>(ClassReader.java:197) ~[asm-9.5.jar:9.5]
	at org.objectweb.asm.ClassReader.<init>(ClassReader.java:180) ~[asm-9.5.jar:9.5]
	at org.objectweb.asm.ClassReader.<init>(ClassReader.java:166) ~[asm-9.5.jar:9.5]
	at org.quiltmc.loader.impl.transformer.QuiltTransformer.transform(QuiltTransformer.java:51) ~[quilt-loader-0.20.1.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPreMixinClassByteArray(KnotClassDelegate.java:511) ~[quilt-loader-0.20.1.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:450) ~[quilt-loader-0.20.1.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:244) ~[quilt-loader-0.20.1.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.loadClassOnly(KnotClassDelegate.java:164) ~[quilt-loader-0.20.1.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:150) ~[quilt-loader-0.20.1.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:239) ~[quilt-loader-0.20.1.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
	at io.netty.util.internal.ObjectPool.newPool(ObjectPool.java:67) ~[netty-common-4.1.82.Final.jar:?]
	at io.netty.buffer.PoolThreadCache$MemoryRegionCache.<clinit>(PoolThreadCache.java:465) ~[netty-buffer-4.1.82.Final.jar:?]
	... 18 more
[15:25:57] [Netty Epoll Server IO #1/DEBUG]: Disabled auto read
[15:25:57] [Netty Epoll Server IO #1/DEBUG]: Enabled auto read

```

**Comment by AlexIIL** [comment]
This turned out to be a problem in `QuiltClassPath::isEqual`, where it didn't check if the input string was shorter than the `Path`, which meant the hash table didn't return the correct path, which caused it to crash while parsing that class.

I'll push a fix soon.

**Comment by AlexIIL** [comment]
This has been fixed in 0.20.2. Thanks for the report!
